vcGL/src/metal/vcTexture.mm

```diff
@@ -328,6 +328,17 @@ bool vcTexture_BeginReadPixels(vcTexture *pTexture, uint32_t x, uint32_t y, uint
 
   @autoreleasepool
   {
+    if (g_pCurrFramebuffer == pFramebuffer)
+    {
+      [g_pCurrFramebuffer->encoder endEncoding];
+      [g_pCurrFramebuffer->commandBuffer commit];
+      [g_pCurrFramebuffer->commandBuffer waitUntilCompleted];
+      g_pCurrFramebuffer->commandBuffer = nil;
+      g_pCurrFramebuffer->encoder = nil;
+      //need this?
```

Comment:
You can probably remove this comment, let's just assume it is needed since this completes the render.